spec/project-spec.coffee

```diff
@@ -197,6 +199,43 @@ describe "Project", ->
           atom.project.bufferForPath("b").then (anotherBuffer) ->
             expect(anotherBuffer).not.toBe buffer
 
+  describe ".repositoryForDirectory(directory)", ->
+    # it() causes beforeEach() in spec-helper.coffee to be run, which reassigns
+    # atom.project. Therefore, the atom.project used in waitsForPromise() will
+    # be different than the atom.project used in a subsequent it(). To work
+    # around this issue, we use runs() instead of it().
+
+    it "there is no repository for /tmp even though there are RepositoryProviders", ->
+      the_result = "dummy_value"
```

Comment:
But the assertion is:

    expect(the_result).toBe null

So that would succeed if the promise did not resolve.